projects/cdk/directives/for-async-queue/for-async-queue.directive.ts

```diff
@@ -0,0 +1,58 @@
+import {
+    Directive,
+    Inject,
+    Input,
+    NgZone,
+    OnDestroy,
+    TemplateRef,
+    ViewContainerRef,
+} from '@angular/core';
+import {TuiDestroyService} from '@taiga-ui/cdk/services';
+import {interval} from 'rxjs';
+import {first, takeUntil} from 'rxjs/operators';
+
+@Directive({
+    selector: '[ngForAsyncQueue][ngForAsyncQueueOf]',
+    providers: [TuiDestroyService],
+})
+export class TuiForAsyncQueueDirective<T> implements OnDestroy {
+    @Input() ngForAsyncQueueTimeout = 10;
+
+    constructor(
+        @Inject(ViewContainerRef) private readonly view: ViewContainerRef,
+        @Inject(TemplateRef) private readonly template: TemplateRef<unknown>,
+        @Inject(NgZone) private readonly ngZone: NgZone,
+        @Inject(TuiDestroyService) private readonly destroy$: TuiDestroyService,
+    ) {}
+
+    @Input()
+    set ngForAsyncQueueOf(items: T[] | readonly T[]) {
+        const queueItems: T[] = items.slice();
+
+        this.asyncQueueWithCreateEmbeddedView(queueItems);
+    }
+
+    asyncQueueWithCreateEmbeddedView(queueItems: T[]): void {
+        if (!queueItems.length) {
+            return;
+        }
+
+        const item: T | undefined = queueItems.shift();
+
+        this.ngZone.runOutsideAngular(() => {
+            interval(this.ngForAsyncQueueTimeout)
+                .pipe(first(), takeUntil(this.destroy$))
+                .subscribe(() => {
+                    this.view
+                        .createEmbeddedView(this.template, {$implicit: item})
+                        .detectChanges();
+
+                    this.asyncQueueWithCreateEmbeddedView(queueItems);
```

Comment:
Can we do it more rxjs way? I mean without recursion, in one stream, with `from(arr), concatMap, delay` ?

Reply:
fixed